### ISSUE TITLE
Create new gpt and partition

### ISSUE THREAD

**Issue by cholcombe973** [Kilobyte22/libpart#1]
I might be reading this api wrongly but it doesn't seem possibly to create a fresh GPT + PartitionEntry in a new file.  Is that correct?

**Comment by Kilobyte22** [comment]
That is correct, this is currently not possible. Shouldn't be too hard to implement, so i'll look into it later today.

**Comment by cholcombe973** [comment]
Yeah I think you have most of the pieces in place. I can lend a hand if you
like.

**Comment by Kilobyte22** [comment]
I don't seem to have the time/motivation right now. If you don't mind, submit a pull request for this and i'll merge it.

**Comment by cholcombe973** [comment]
Ok no problem.

**Comment by Kilobyte22** [comment]
Small status update: As of commit 76af974 it should be possible to create a GPT using `GPTTable::new(...)`. Furthermore, there is the `GPTTable::set_partition(&mut self, ...)` function for modifying the table. Creating a `PartitionEntry` is possible using either the `::default()` method or manually filling the entry. Please note that this API is not very well tested (and certainly not stabilized) yet.

**Comment by cholcombe973** [comment]
Awesome!  Thanks
